codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -236,7 +236,8 @@ private void writePendingBytes() {
 
     /**
      * This will allocate bytes by stream weight and priority for the entire tree rooted at {@code parent}, but does
-     * not write any bytes.
+     * not write any bytes. Priority is generally distributed proportionally to siblings in the priority tree, however
+     * we ensure that at least 1 byte is assigned to each stream in order to make progress.
```

Comment:
@buchgr - My concern is the current approach will allocate "fairly" according to weight `+/-` some rounding (most likely just `1` across a few streams) and completely use the available connection window.  If you can open another issue and detail your suggestion with respect to connection window, stream window, sibling weights that would be a good place to start the discussion.